openshift/openshift-azure
# e2e test refactor take 2

Building on @charlesakalugwu 's work in #762.

* merge test/e2erp into test/e2e
* move tests into 1 suite and 3 packages for now:
  - test/e2e/specs (general sanity checking)
  - test/e2e/specs/fakerp (stuff which needs system:admin and hence can't yet work in prod)
  - test/e2e/specs/realrp (stuff which needs prod RP RBAC and hence can't work with the plugin)
* add login functionality to remove KUBECONFIG and oc login hacks in e2e.sh (see test/clients/openshift/login.go)
* put all standard `func() (bool, error)` functions used for readiness checking into single package pkg/util/ready (TODO: more test cases)
* remove random name generator copy/pasted from k8s and generalise config code to pkg/util/randomstring
* grudgingly add an official combined azure test client and combined openshift test client (test/clients/{azure,openshift}) with as few helper functions as I think I can get away with
* quite a lot of test clean-up and simplification

@charlesakalugwu @kargakis @mjudeikis @asalkeld ptal.
